util/awsservice/cloudwatchlogs.go

```diff
@@ -197,6 +197,47 @@ func GetLogQueryStats(logGroupName string, startTime, endTime int64, queryString
 	}
 }
 
+// GetLogQueryResults for the log group between start/end (in epoch seconds) for the
+// query string.
+func GetLogQueryResults(logGroupName string, startTime, endTime int64, queryString string) ([][]types.ResultField, error) {
```

Comment:
we don't have something like this already? Something that checks for the existence of logs in a log group within a certain time period?

Reply:
We have [GetLogQueryStats](https://github.com/aws/amazon-cloudwatch-agent-test/blob/eks-entity/util/awsservice/cloudwatchlogs.go#L161) function which does what you described but that returns the statistics of the query result which does not include log fields so we can't validate entity from that. `GetLogQueryResults` function is a copy of that function except that it returns `results.Results`